baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/HandleInterruption.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.Matchers;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.google.errorprone.util.ErrorProneTokens;
+import com.sun.source.tree.BlockTree;
+import com.sun.source.tree.CatchTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.IdentifierTree;
+import com.sun.source.tree.InstanceOfTree;
+import com.sun.source.tree.StatementTree;
+import com.sun.source.tree.ThrowTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.TryTree;
+import com.sun.tools.javac.code.Type;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.type.UnionType;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "HandleInterruption",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = BugPattern.SeverityLevel.WARNING,
+        summary = "InterruptedException must be handled by either rethrowing the InterruptedException or setting "
+                + "Thread.currentThread().interrupt(). Failure to do so can result in retrial of long-running "
+                + "operations that are expected to be terminated.")
```

Comment:
Need to reference the `// interruption reset` suppression in the summary

Reply:
Should link https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html (Brian Goetz)

Reply:
@iamdanfox I'm not sure we want to limit ourselves to cases that rethrow -- those are the most likely to do the _right_ thing. The bad case is when InterruptedException is caught, the interrupted flag is not reset, and execution continues.